### PR TITLE
support data cache for webdataset

### PR DESCRIPTION
**Pull request by chenyang23333**
# What does this PR do?

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/stable_diffusion_xl/gm/data/dataset_wds.py [hunk 1]
```diff
@@ -131,32 +147,20 @@ def __init__(
...
-            )
-
-            caption = str(caption)
-
-            if _is_valid_text_input(caption):
-                print("WARNING: convert caption type to string success.", flush=True)
-            else:
-                caption = " "
-                print("WARNING: convert caption type to string fail, set caption to ` `.", flush=True)
-
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:159]: 这个代码为什么删除了？

File: examples/stable_diffusion_xl/gm/data/dataset_wds.py [hunk 2]
```diff
@@ -196,16 +205,13 @@ def collate_fn(self, samples, batch_info):
...
 
         if self.tokenizer:
-            data = {k: (v.tolist() if k == "txt" else v.astype(np.float32)) for k, v in data.items()}
-
-            try:
-                tokens, _ = self.tokenizer(data)
-            except Exception as e:
-                print(f"WARNING: tokenize fail, error mg: {e}, convert data[`txt`]: {data['txt']} to ` `", flush=True)
-                data["txt"] = [" " for _ in range(len(data["txt"]))]
-                tokens, _ = self.tokenizer(data)
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:206]: 这个代码为什么删除了？

File: examples/stable_diffusion_xl/gm/data/dataset_wds.py [hunk 3]
```diff
@@ -172,6 +176,11 @@ def preprocess(self, image, caption: str):
             ),
         }
 
+        if self.return_sample_name:
+            self.dataset_output_column_names.append("sample_name")
+            sample["sample_name"] = np.array(image_path)
+            print("sample_name is :", image_path)
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:182]: 不用每个step都print

File: examples/stable_diffusion_xl/gm/data/dataset_wds.py [hunk 4]
```diff
@@ -329,19 +358,19 @@ def __init__(self, shardlist_desc=None, *args, **kwargs):
         for raw in self.wds_iterator:
             try:
                 image, caption = self.parse_raw_data(raw)
-                sample = self.preprocess(image, caption)
+                sample = self.preprocess(image, caption, str(raw["__key__"]))
```
Comment (geniuspatrick) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:361]: `getattr`

File: examples/stable_diffusion_xl/gm/data/dataset_wds.py [hunk 5]
```diff
@@ -88,12 +89,16 @@ def __init__(
         per_batch_size=1,  # for multi_aspect
         caption_key="caption",
         prompt_empty_probability=0.0,
+        return_sample_name=True,
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:92]: `return_sample_name` default shuold be `False`

File: examples/stable_diffusion_xl/gm/data/dataset_wds.py [hunk 6]
```diff
@@ -385,15 +415,17 @@ class T2I_Webdataset_RndAcs(T2I_BaseDataset):
     def __init__(self, shardlist_desc=None, cache_dir=None, *args, **kwargs):
         # shardlist_desc: path to a json file describing sample num for each tar
         super().__init__(*args, **kwargs)
+        self.data_path = kwargs.get("data_path")
+        self.return_sample_name = kwargs.get("return_sample_name")
         if shardlist_desc is None:
             data_path = kwargs.get("data_path")
             if not os.path.exists(os.path.join(data_path, "data_info.json")):
-                print("Scanning tar files to get sample nums...")
+                print("Scanning tar files to get sample nums...", flush=True)
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:423]: flush相关的改动去掉

File: examples/stable_diffusion_xl/gm/data/dataset_wds.py [hunk 7]
```diff
@@ -385,15 +415,17 @@ class T2I_Webdataset_RndAcs(T2I_BaseDataset):
     def __init__(self, shardlist_desc=None, cache_dir=None, *args, **kwargs):
         # shardlist_desc: path to a json file describing sample num for each tar
         super().__init__(*args, **kwargs)
+        self.data_path = kwargs.get("data_path")
+        self.return_sample_name = kwargs.get("return_sample_name")
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:419]: self.return_sample_name = kwargs.get("return_sample_name")
=>
self.return_sample_name = kwargs.get("return_sample_name", False)

File: examples/stable_diffusion_xl/gm/data/dataset_wds.py [hunk 8]
```diff
@@ -444,14 +477,16 @@ def __getitem__(self, idx):
             raw = self.dataset[idx]
             image, caption = self.parse_raw_data(raw)
             sample = self.preprocess(image, caption)
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:479]: image name?

Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:479]: __init__中的preprocess未考虑

**Pull request closed without merging**